core/src/main/scala/org/apache/spark/util/collection/BitSet.scala

```diff
@@ -58,6 +53,19 @@ class BitSet(numBits: Int) extends Serializable {
   }
 
   /**
+   * Clear all the bits up to a given index
+   */
+  def clearUntil(bitIndex: Int): Unit = {
+    val wordIndex = bitIndex >> 6 // divide by 64
+    Arrays.fill(words, 0, wordIndex, 0)
+    if(wordIndex < words.length) {
+      // Set the remaining bits
```

Comment:
Well, that was dumb.  Sorry and thanks.  I can update the PR in about seven hours if you don't merge it today :shrug: